app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperState.kt

```diff
@@ -20,4 +25,29 @@ data class WallpaperState(
             availableWallpapers = listOf(),
         )
     }
+
+    /**
+     * Helper property used to obtain the [Color] to fill-in cards in front of a wallpaper.
+     *
+     * @return The appropriate light or dark wallpaper card [Color], if available, otherwise a default.
+     */
+    val wallpaperCardColor: Color
+        @Composable get() = when {
+            areWallpaperCardColorsAvailable() -> {
+                if (isSystemInDarkTheme()) {
+                    Color(currentWallpaper.cardColorDark!!)
+                } else {
+                    Color(currentWallpaper.cardColorLight!!)
+                }
+            }
+            else -> FirefoxTheme.colors.layer2
+        }
+
+    /**
+     * Helper function used to determine if the wallpaper card colors are available for use.
+     *
+     * @return true if both wallpaper card colors are not null
+     */
+    fun areWallpaperCardColorsAvailable(): Boolean =
+        currentWallpaper.cardColorLight != null && currentWallpaper.cardColorDark != null
```

Comment:
Some of the awkwardness in calling `areWallpaperCardColorsAvailable`  could potentially be avoided by a data class or a typealias. For example:

```
data class WallpaperCardColors(val light: Color, val dark: Color)

val wallpaperCardColors: WallpaperCardColors
    @Composable get() = {
        Result.runCatching {
            WallpaperCardColors(currentWallpaper.cardColorLight!!, currentWallpaper.cardColorDark!!)
        }.getOrDefault(WallpaperCardColors(<default light>, <default dark>))
    }
```

Reply:
I see the `View` style usages of this style require `Int`s. So I would consider having the properties of the data class remain `Long`s and add extension functions to convert them to appropriate types, so you could do something like:

```
wallpaperCardColors.light.toViewColor
```
and
```
wallpaperCardColors.light.toComposeColor
```

maybe also even this, to avoid the conditionals sprinkled throughout:
```
wallpaperCardColors.toSystemThemeColor(isSystemInDarkMode())
```

Reply:
Re: suggestion 1

I'd have to forcibly unwrap the underlying colors of `FirefoxTheme.colors.layer2`. Usually we try to avoid this, but I know that the photon colors are directly used in `RecentBookmarks` and `RecentTabs` but that's because, IIRC, they don't have direct color token mapping.

Separately, there's one or two components (`CustomizeHomeButtonViewHolder`) that rely on the check and the colors without using the fallback color. 

I'd be down to pair on this pair on this though; I agree this was a simplification to a clunky use case.

Reply:
Re: suggestion 2

The color adaptation to a View-safe consumption is only a problem for Top Sites, as this is the last component on the homepage to be converted to Compose, so I'm not sure the juice is worth the squeeze here. 

I don't think I'm 100% understanding the `toSystemThemeColor` suggestion.